python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py

```diff
@@ -179,70 +279,68 @@ def configure(self, masterappconfig):
         """
         # Lets test the inputs
         opt_file = self.getOptsFile()
-        dir_name = self.getDir()
+        dir_name = self.directory
         return (None, None)
 
     def getOptsFile(self):
         """
-        This function returns a sanitized absolute path to the self.myOpts file from user input
+        This function returns a sanitized absolute path to the self.options file from user input
         """
-        if self.myOpts:
-            if isinstance(self.myOpts, LocalFile):
+        if self.options:
+            if isinstance(self.options, LocalFile):
                 ## FIXME LocalFile should return the basename and folder in 2 attibutes so we can piece it together, now it doesn't
-                full_path = path.join(self.myOpts.localDir, self.myOpts.namePattern)
+                full_path = path.join(self.options.localDir, self.options.namePattern)
                 if not path.exists(full_path):
                     raise ApplicationConfigurationError(None, "Opts File: \'%s\' has been specified but does not exist please check and try again!" % full_path)
-                return self.myOpts
-            elif isinstance(self.myOpts, DiracFile):
-                return self.myOpts
+                return self.options
+            elif isinstance(self.options, DiracFile):
+                return self.options
             else:
-                raise ApplicationConfigurationError(None, "Opts file type %s not yet supported please contact Ganga devs if you require this support" % getName(self.myOpts))
+                raise ApplicationConfigurationError(None, "Opts file type %s not yet supported please contact Ganga devs if you require this support" % getName(self.options))
         else:
             raise ApplicationConfigurationError(None, "No Opts File has been specified, please provide one!")
 
-    def getDir(self):
+    def getEnvScript(self):
         """
-        This function returns a sanitized absolute path of the self.directory method from user input
+        Return the script which wraps the running command in a correct environment
         """
-        if self.directory:
-            return path.abspath(expandfilename(self.directory))
-        else:
-            raise ApplicationConfigurationError(None, "No Opts File has been specified, please provide one!")
-
-    def getEnvScript(self):
         return 'export CMTCONFIG=%s; source LbLogin.sh --cmtconfig=%s && ' % (self.platform, self.platform)
 
-    def exec_cmd(self, cmd):
+    def execCmd(self, cmd):
         """
         This method executes a command within the namespace of the project. The cmd is placed in a bash script which is executed within the env
+        This will adopt the platform associated with this application and all commands requiring to be run within the project env
+        will be required to be executed with the './run' explcitly called
```

Comment:
Give an explicit example or two here. Presumably `j.application.execCmd('make')` is the most obvious one.